### PR TITLE
GAIA-207 | Updated Dialog and Text components:

### PR DESCRIPTION
**Pull request by vladstart1**
* Split `Dialog` onto smaller components
* Fixed `Text`'s styles
* Removed `DialogButton`
* Updated typescript package

### CODE REVIEW

File: src/components/Dialog/DialogTitle/index.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import useStyles from './styles';
+import Text, {ITextProps} from '../../Text';
+
+interface IDialogTitle extends ITextProps {}
+
+const DialogTitle: React.FC<IDialogTitle> = ({children, style, ...rest}) => {
+  const styles = useStyles();
+  return (
+    <Text style={[styles.dialogTitle, style]} variant="h4" {...rest}>
```
Comment (AlinaKuzmenko) [inline src/components/Dialog/DialogTitle/index.tsx:10]: what if we make children: string | ReactNode?

File: src/components/Dialog/DialogActions/index.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+interface IDialogActions extends ViewProps {}
+
+const DialogActions: React.FC<IDialogActions> = ({
+  children,
+  style,
+  ...rest
+}) => {
+  const styles = useStyles();
+  return (
+    <View style={[styles.actionsWrapper, style]} {...rest}>
```
Comment (AlinaKuzmenko) [inline src/components/Dialog/DialogActions/index.tsx:14]: please, name the className the same as the component (dialogActions)

File: src/components/Dialog/Dialog.stories.tsx [hunk 3]
```diff
@@ -21,12 +22,23 @@ const DialogStory = (args: IDialogProps) => {
...
       <Button onPress={() => setOpen(true)}>Open Dialog</Button>
-      <Dialog
-        {...args}
-        onCancel={closeDialog}
-        onConfirm={closeDialog}
-        open={open}
-      />
+      <Dialog {...args} onCancel={closeDialog} open={open}>
+        <Dialog.Title>Discard draft?</Dialog.Title>
+        <Text variant="body">Discard draft?</Text>
```
Comment (AlinaKuzmenko) [inline src/components/Dialog/Dialog.stories.tsx:27]: On the web, we have a DialogContent component for modal content

File: src/components/Dialog/index.tsx [hunk 4]
```diff
@@ -1,28 +1,25 @@
...
 
 export interface IDialogProps extends ModalProps {
-  title?: string;
-  text?: string | React.ReactElement;
   onCancel: () => void;
-  onConfirm: () => void;
-  cancelButtonText?: string;
-  confirmButtonText?: string;
   open: boolean;
   cancelOnOutsidePress?: boolean;
```
Comment (AlinaKuzmenko) [inline src/components/Dialog/index.tsx:10]: `cancelOnOutsidePress` was renamed to `disableOutsideClick` in the web library

File: src/components/Dialog/index.tsx [hunk 5]
```diff
@@ -1,28 +1,25 @@
...
-  title,
-  text,
+export interface IDialogStaticProps {
+  Title: typeof DialogTitle;
+  Actions: typeof DialogActions;
+}
+
+const Dialog: FC<IDialogProps> & IDialogStaticProps = ({
   open,
   onCancel,
```
Comment (AlinaKuzmenko) [inline src/components/Dialog/index.tsx:20]: `onCancel` was renamed to `onClose` in the web library

File: src/components/Dialog/index.tsx [hunk 6]
```diff
@@ -1,28 +1,25 @@
-import React from 'react';
+import React, {FC} from 'react';
 import useStyles from './styles';
-import {Modal, Pressable, View, Text, ModalProps} from 'react-native';
-import DialogButton from './DialogButton';
+import {Modal, Pressable, ModalProps} from 'react-native';
+import DialogTitle from './DialogTitle';
+import DialogActions from './DialogActions';
 
 export interface IDialogProps extends ModalProps {
```
Comment (AlinaKuzmenko) [inline src/components/Dialog/index.tsx:7]: and we also have an `overlayClassName` in the web library

**Review by AlinaKuzmenko — CHANGES_REQUESTED** [review]

File: src/components/Dialog/DialogActions/index.tsx (hunk 2 above)
Comment (vladstart1) [reply src/components/Dialog/DialogActions/index.tsx:14]: React native doesn't need `className` as it uses styles not `classNames`, so I don't see any reason to use the same name for styles

File: src/components/Dialog/DialogTitle/index.tsx (hunk 1 above)
Comment (vladstart1) [reply src/components/Dialog/DialogTitle/index.tsx:10]: It doesn't really make sense to do so. If someone would need custom component he can use his component instead `Dialog.Title`. It's not mandatory to use `Dialog.Title` inside `Dialog`

File: src/components/Dialog/Dialog.stories.tsx (hunk 3 above)
Comment (AlinaKuzmenko) [reply src/components/Dialog/Dialog.stories.tsx:27]: You're right, we don't actually need it. Then we need to remove it from the web library

File: src/components/Dialog/DialogActions/index.tsx (hunk 2 above)
Comment (AlinaKuzmenko) [reply src/components/Dialog/DialogActions/index.tsx:14]: Just to be consistent with the way we name components and their styles in web and RN libraries

**Pull request merged**